### PR TITLE
proposal to support gz and xz compression

### LINKED ISSUE #29: Support for decompressing jdk.tar.xz
If you specify this as URL: https://builds.shipilev.net/openjdk-shenandoah-jdk8/openjdk-shenandoah-jdk8-latest-x86_64-release.tar.xz it fails via

> gzip: stdin: not in gzip format

But when using tar ala:

`tar -xf file.tar.xz`

it works, as recent versions auto detect the compression. Or one can use the explicit flag `-J` and auto detect the ending in the script.

### PR DESCRIPTION
**Pull request by karussell**
Fixes #29

### CODE REVIEW

**Comment by sormuras** [comment]
Cancelled the pending Travis CI build as it won't be successful due to #27 
The "untar" of the "gz" worked on Linux and Mac OS, thus merging this change.

**Comment by karussell** [comment]
Ah, ok. Thanks for handling this so fast!

**Comment by sormuras** [comment]
The last job in the last stage should work now.

https://travis-ci.org/sormuras/sormuras.github.io/builds/416835140

File: install-jdk.sh [hunk 1]
```diff
@@ -248,7 +248,7 @@ function download_and_extract_and_set_target() {
     local local="--directory-prefix ${workspace}"
     local remote='--timestamping --continue'
     local wget_options="${quiet} ${local} ${remote}"
-    local tar_options="-z --file ${archive}" # "--auto-compress" is not supported on mac osx, using "-z"
+    local tar_options="--file ${archive}" # "--auto-compress" is not supported on mac osx, using "-z"
```
Comment (Stephan202) [inline install-jdk.sh]: The comment is now out-of-sync with the code, no?

Comment (sormuras) [reply install-jdk.sh]: Aye. Removing it.

Comment (karussell) [reply install-jdk.sh]: Sorry, should have done this but I did not understand it - now I do ;)

**Pull request merged**